cfme/infrastructure/provider/kubevirt.py

```diff
@@ -0,0 +1,47 @@
+from cfme.utils.providers import get_crud
+from cfme.containers.provider.openshift import VirtualizationEndpoint
+from . import InfraProvider
+
+
+class KubeVirtProvider(InfraProvider):
+
+    def __init__(self, parent_provider, **kwargs):
+
+        self.parent_provider = parent_provider
+
+        super(KubeVirtProvider, self).__init__(
+            name=kwargs.get('name'),
+            endpoints=kwargs.get('endpoints'),
+            key=kwargs.get('key'),
+            provider_data=kwargs.get('provider_data'),
+            appliance=kwargs.get('appliance'))
+
+    def create(self, *args, **kwargs):
+        # Override the standard behaviour to actually create the parent container provider first.
+        created = self.parent_provider.create()
+        return created
```

Comment:
**QUESTION**: If you say "create parent provider first" then where the actual provider gets created? Isn't there a super call missing? By the way you don't need an intermediate variable in here if this is all it gets.

Reply:
@mfalesni the `KubeVirt(InfraProvider)` gets automatically created on the CF side when we add the ocp with `KubeVirt` virtualization

Reply:
@vatsalparekh I think a more descriptive comment in the code avoid potential confusion here.  Instead of saying create the provider first - state the KubeVirt provider is added automatically when creating the parent Openshift Provider, and no further action is needed to see this provider.